### PR TITLE
Add http errors

### PR DESCRIPTION
**Pull request by pierroberto**
See #52

### CODE REVIEW

File: src/Http.ts [hunk 1]
```diff
@@ -19,9 +19,46 @@ import * as $T from './Type'
...
+  MethodNotAllowed: 405,
+  NotAcceptable: 406,
+  ProxyAuthenticationRequired: 407,
+  RequestTimeout: 408,
+  Conflict: 409,
+  Gone: 410,
+  LengthRequired: 411,
+  PreconditionFailed: 412,
+  PayloadTooLarge: 413,
+  RequestURITooLong: 414,
```
Comment (xzhavilla) [inline src/Http.ts:34]: Pascal case makes acronyms work as normal words:
```suggestion
  RequestUriTooLong: 414,
```

File: src/Http.ts [hunk 2]
```diff
@@ -19,9 +19,46 @@ import * as $T from './Type'
...
+  RequestHeaderFieldsTooLarge: 431,
+  ConnectionClosedWithoutResponse: 444,
+  UnavailableForLegalReasons: 451,
+  ClientClosedRequest: 499,
+  InternalServerError: 500,
+  NotImplemented: 501,
+  BadGateway: 502,
+  ServiceUnavailable: 503,
+  GatewayTimeout: 504,
+  HTTPVersionNotSupported: 505,
```
Comment (xzhavilla) [inline src/Http.ts:55]: Pascal case, again:
```suggestion
  HttpVersionNotSupported: 505,
```

**Review by xzhavilla — CHANGES_REQUESTED** [review]

**Review by xzhavilla — APPROVED** [review]

**Comment by xzhavilla** [comment]
@pierroberto: is this ready to go?

**Comment by xzhavilla** [comment]
@pierroberto: feel free to merge when ready.

**Comment by pierroberto** [comment]
@xzhavilla I updated the changelog. It's ready to be merged.

**Pull request merged**